### PR TITLE
[approvaltests.cpp] Remove default options from test package

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **ApprovalTests.CPP/any**

* Remove default_options from test_package
* Update Conan recipe according the conan-center hook
* Check compiler version under validate method

Related to https://github.com/conan-io/hooks/issues/287

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
there are many changes... and we are no longer testing all the backends... I am tentative

File: recipes/approvaltests.cpp/all/test_package/test_package.cpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <cstdlib>
+#include <iostream>
+#include "ApprovalTests.hpp"
+
+
+int main(void) {
+    std::cout << ApprovalTests::StringMaker::toString(42) << " is the answer."<< std::endl;
```
Comment (ericLemanissier) [inline recipes/approvaltests.cpp/all/test_package/test_package.cpp:7]: this is risky. We would better use a symbol from ApprovalTests library, or the linker will discard the object file, and we have no chance of detecting missing requirements/system requirements.

File: recipes/approvaltests.cpp/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#include <cstdlib>
+#include <iostream>
+#include "ApprovalTests.hpp"
+
```
Comment (ericLemanissier) [inline recipes/approvaltests.cpp/all/test_package/test_package.cpp:4]:
```suggestion
#include "utilities/Grid.h"

```

**Comment by uilianries** [comment]
> there are many changes... and we are no longer testing all the backends... I am tentative

We are not testing the default package

File: recipes/approvaltests.cpp/all/test_package/test_package.cpp (hunk 1 above)
Comment (ericLemanissier) [inline recipes/approvaltests.cpp/all/test_package/test_package.cpp:7]:
```suggestion
    std::cout << ApprovalTests::StringMaker::toString(42) << ApprovalTests::Grid::print(1, 1, " is the answer.")<< std::endl;
```
ApprovalTests::Grid::print is not inline, so we are sure that the library is not discarded

**Review by ericLemanissier — COMMENTED** [review]
ApprovalTests::Grid::print is not inline so

**Comment by uilianries** [comment]
@ericLemanissier Thank you! Well observed!

File: recipes/approvaltests.cpp/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#include <cstdlib>
+#include <iostream>
+#include "ApprovalTests.hpp"
+#include "utilities/Grid.h"
```
Comment (ericLemanissier) [inline recipes/approvaltests.cpp/all/test_package/test_package.cpp:4]:
```suggestion
#include "ApprovalTests/utilities/Grid.h"
```

**Comment by uilianries** [comment]
@ericLemanissier I needed to revert that change. `Grid` is a new feature, not available on previous versions

**Comment by ericLemanissier** [comment]
Yes, I saw that. Approval was header only in early versions anyway

**Comment by prince-chrismc** [comment]
CI looks stuck

**Comment by SSE4** [comment]
> CI looks stuck

it shouldn't block, it is just a secondary notification didn't arrive

**Review by prince-chrismc — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by ericLemanissier — CHANGES_REQUESTED** [review]
no test is actually run with this version of the PR. You may simply want to add back the small test you had in a previous version of this PR ?

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**